src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -1669,12 +1669,10 @@ impl<'tcx> fmt::Debug for MissingConstructors<'tcx> {
     }
 }
 
-// TODO: we should never need that and it's very expensive
+/// The implementation panics because this should not happen
 impl<'tcx> PartialEq<Self> for MissingConstructors<'tcx> {
```

Comment:
It would be good to add a comment saying why it is necessary (i.e. because it's nested inside `Constructor`).